src/occwl/shape.py

```diff
@@ -422,7 +422,7 @@ def transform(self, a, copy=True):
         # we don't want to set the values as this
         # would give us a geometric identity without
         # the identity flag set
-        if not np.allclose(a, np.eye(4)):
+        if not np.allclose(a, np.eye(3, 4)):

```

Comment:
I see.   Yes.   This shows a definite problem with my code.  A homogeneous transform matrix is usually 4x4.  The bottom row is `[0 0 0 1]`.   The comment at the top of the method says that a 4x4 matrix is expected, but the values of the bottom row are never checked and just ignored.

I suggest we do the following

- Change the comment at the top of the function
- Assert that the shape of the matrix passed to the function is correct

Reply:
Thanks - didn't see that in the doc-string, saw the algebra and stopped there.